docs/exceptions.md

```diff
@@ -192,12 +192,11 @@ Do:
 * create a custom exception type
 * use `throwM` (`MonadThrow`)
 
-If you want to return `m (Either e a)` from a function, it's
-recommended to define `instance TypeError "NOT AN EXC" => Exception e`
-for your type `e`. If the meaning of `e` type is not related to
-exceptional situations at all, it's not needed to define such
-instance. But for example if `e` denotes a `ParseError`, please do
-define it.
+If you want to return `m (Either e a)` from a function or to use `ExceptT e m
+a`, it's required to define `instance TypeError "NOT AN EXC" => Exception e` for
+your type `e`. Do not use `ExceptT e m a` in exported top-level functions,
+convert to `m (Either e a)` using `runExceptT`. As of now, using `ExceptT` robs
+us of `bracket`, but this will be fixed in the next release of `exceptions`.
```

Comment:
Could you explain, please, how this will be fixed in `exceptions` package? Maybe I'm missing something, but AFAIU, `ExceptT` can't have `MonadMask` instance and `bracket` can't be used with `ExceptT`.

* https://github.com/ekmett/exceptions/issues/35

Though, in CHANGELOG I see that there will be `MonadMask` instance for `ExceptT`...

* https://github.com/ekmett/exceptions/blob/master/CHANGELOG.markdown#090

I'm confused 😕

Reply:
This is already fixed in `master` by adding `generalBracket` to the `MonadMask` class. `ExceptT` couldn't have a `MonadMask` instance before, but now it can.

Reply:
I'm curious about `generalBracket`. Where is it defined? Are you referring to `master` of cardano-sl?

Reply:
I am referring to `master` of the `exceptions` package, where `MonadMask` is defined.